### DISCUSSION TITLE
https://adyanth.site/posts/migration-compose-k8s/nvidia-graphics-kubernetes/

### DISCUSSION THREAD

**Discussion by giscus** [adyanth/blog-comments#17] [General]
# https://adyanth.site/posts/migration-compose-k8s/nvidia-graphics-kubernetes/

A guide out of my struggle to get consumer NVIDIA graphics available to workloads running on K3s

https://adyanth.site/posts/migration-compose-k8s/nvidia-graphics-kubernetes/

**Comment by spookyuser** [comment]
Wow super interesting! I'm a k8s noob so forgive me if the answer is obvious, but would it be possible to have an nvidia node that scales itself down to zero after handling requests to take advantage of per minute billing on something like aws?

**Reply by adyanth** [reply]
Yeah, cluster autoscalers do that, although requesting a node with GPU is something I haven't seen. If it does not exist already (I'm sure someone would have written it), custom controllers can do this too.

Reference: https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler

**Reply by spookyuser** [reply]
Thanks for the info! I'll check that out, I'm almost definitely prematurely optimising something by trying to achieve this haha.
